src/runtime/runtime_unix.go

```diff
@@ -222,46 +224,30 @@ func nanosecondsToTicks(ns int64) timeUnit {
 }
 
 func sleepTicks(d timeUnit) {
-	// When there are no signal handlers present, we can simply go to sleep.
-	if !hasSignals {
-		// timeUnit is in nanoseconds, so need to convert to microseconds here.
-		usleep(uint(d) / 1000)
-		return
+	var until timeUnit
+	if !hasScheduler {
+		until = ticks() + d
```

Comment:
I don't understand why this code depends on `hasScheduler`.

Reply:
Yeah, that's a bug. Thanks for spotting it! (I probably modified the code a few times and it probably made sense to not calculate `until` without a scheduler before).
I'll push a fix.